Sources/MapboxCoreNavigation/NavigationService.swift

```diff
@@ -181,7 +181,7 @@ public class MapboxNavigationService: NSObject, NavigationService {
         delegate?.navigationService(self, willBeginSimulating: progress, becauseOf: intent)
         announceSimulationDidChange(.willBeginSimulation)
         
-        simulatedLocationSource = SimulatedLocationManager(routeProgress: progress)
+        simulatedLocationSource = simulatedLocationSourceType.init(route: progress.route, currentDistance: 0, currentSpeed: 0)
```

Comment:
This change looses `currentDistance` calculation it had before. Without it, if simulation start in the middle of the route it will jump back to the origin instead.

Reply:
Thanks a lot, I messed the constructors. I fixed that and add tests for checking distance